src/js/view/components/nextuptooltip.js

```diff
@@ -0,0 +1,238 @@
+
+            }
+
+            // Set header
+            this.header = this.content.getElementsByClassName('jw-nextup-header')[0];
+            this.header.innerText = this.nextUpText;
+
+            // Set title
+            this.title = this.content.getElementsByClassName('jw-nextup-title')[0];
+            this.title.innerText = nextUpItem.title || '';
+
+        },
+        onPlaylistItem: function(item) {
+            // Listen for duration changes to determine the offset
+            // for when next up should be shown
+            this._model.on('change:duration', this.onDuration, this);
+
+            var playlist = this._model.get('playlist');
+            var nextUpIndex = (item.index + 1) % playlist.length;
+            var nextUpItem = playlist[nextUpIndex];
+            this.relatedMode = false;
+            this.setNextUpItem(nextUpItem);
+
+            this._related = this._related || this._api.getPlugin('related');
+            if (this._related && item.index === playlist.length - 1) {
+                // Only switch to related mode if there is a related playlist
+                this._related.on('playlist', this.onRelatedPlaylist.bind(this));
+            }
+        },
+        onDuration: function(model, duration) {
+            if (!duration) {
+                return;
+            }
+            // Use nextupoffset if set or default to 10 seconds from the end of playback
+            var offset = model.get('nextupoffset') || -10;
+            offset = utils.seconds(offset);
+
+            if (offset < 0) {
+                // Determine offset from the end. Duration may change.
+                offset += duration;
+            } else {
+                // Offset is from the beginning of playback.
+                // No need to listen for further duration changes.
+                model.off('change:duration', this.onDuration, this);
+            }
+
+            this.offset = offset;
+        },
+        onRelatedPlaylist: function(evt) {
+            if (evt.playlist && evt.playlist.length) {
+                this.relatedMode = true;
+                this.setNextUpItem(evt.playlist[0]);
+            }
+        },
+        onMediaModel: function(model, mediaModel) {
+            mediaModel.on('change:state', function(model, state) {
+                if (state === 'complete') {
+                    this.state = 'tooltip';
+                    this.hide();
+                }
+            }, this);
+        },
+        onElapsed: function(model, val) {
+            // Show nextup if:
+            // - in playlist mode but not playing an ad
+            // - autoplaying in related mode and autoplaytimer is set to 0
+            if (this.showNextUp && (val >= this.offset)) {
+                this.showTilEnd();
+            } else if (this.state === 'opened' || this.state === 'closed') {
+                this.state = 'tooltip';
+                this.hide();
+            }
+        },
+        element: function() {
```

Comment:
@johnBartos It's used in the view: 
`_controlsLayer.appendChild(_nextuptooltip.element());`